pkg/vault/contracts/BatchRouter.sol

```diff
@@ -157,7 +157,6 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
             if (path.steps[0].isBuffer && EVMCallModeHelpers.isStaticCall() == false) {
                 // If first step is a buffer, take the token in advance. We need this to wrap/unwrap.
                 _takeTokenIn(params.sender, stepTokenIn, stepExactAmountIn, false);
-                _settledTokenAmounts().tAdd(address(stepTokenIn), stepExactAmountIn);
```

Comment:
This was leaving some dust if batch router is used again in the same transaction, and the settled amount is not used.

Reply:
This line should also go away if I'm not mistaken: https://github.com/balancer/balancer-v3-monorepo/pull/785/files#diff-3bcb4ada7438e6f8075537f3930a0099ae12d4389c22e6cb86c973f84681b0e7R456

Settled token amounts is only for calculated amounts, not for given inputs.

Reply:
@jubeira , about this, when I remove this line the test breaks. In this case, tokenIn is calculated and user paid maxAmountsIn in advance. We need the settledTokenAmounts for tokenIn to return the difference between maxAmountsIn and amountsIn at the end of the operation.

Reply:
Got it; makes sense.